nilearn/region_extraction/regionextractor.py

```diff
@@ -0,0 +1,232 @@
+"""
+region_extractor
+"""
+
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+
+from skimage.feature import peak_local_max
+from skimage.segmentation import random_walker
+
+from nilearn import plotting
+from nilearn._utils import check_niimg
+from nilearn._utils.extmath import fast_abs_percentile
+from nilearn.image import iter_img, new_img_like
+from nilearn.image.image import _smooth_array
+from nilearn._utils.niimg_conversions import concat_niimgs
+from nilearn._utils.compat import _basestring
+
+
+def extract_seperate_regions(img, img_data, min_size):
+    """ Extract and seperate the regions which have largest
+        connections considering group of voxels which are
+        certainly high given a thresholded image of the
+        brain maps/components.
+
+    Parameters
+    ----------
+    img: 3D Nifti like image/object
+        contains a thresholded image of the components.
+
+    Returns
+    -------
+    regions_img: a 4D Nifti like image/object
+        contains the regions extracted from
+        the input image where each 3D volume is
+        assigned as each seperate region by
+        concatenating in the 4th dimension to a 4D image.
+    """
+    regions_img = []
```

Comment:
Same remark as before. Suffix `img` is for images object (ie `SpatialImage` with `get_data` and `get_affine` methods).